api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+// const { Client } = require('pg');
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const token = req.headers['x-access-token'];
+  if (!token) {
+    throw new ClientError(401, 'authentication required');
+  }
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
```

Comment:
You need to call `next(err)` here so the code will go to the Error handle middleware then send a response to the client. Without calling next here the server would hang and never send any response back to the client.